Source/WebCore/Modules/webauthn/AuthenticatorCoordinator.cpp

```diff
@@ -272,6 +291,16 @@ void AuthenticatorCoordinator::discoverFromExternalSource(const Document& docume
         ASSERT(!exception.message.isNull());
         promise.reject(exception.toException());
     };
+
+    if (m_isCancelling) {
+        m_queuedRequest = [weakThis = WeakPtr { *this }, weakFrame = WeakPtr { *frame }, weakDocument = WeakPtr<Document, WeakPtrImplWithEventTargetData> { document }, clientDataJsonHash = WTFMove(clientDataJsonHash), requestOptions = WTFMove(requestOptions), &scopeAndCrossOriginParent, callback = WTFMove(callback)]() mutable {
+            if (!weakThis || !weakFrame || !weakDocument)
+                return;
+            const auto& options = requestOptions.publicKey.value();
+            weakThis->m_client->getAssertion(*weakFrame, weakDocument->securityOrigin(), clientDataJsonHash, options, requestOptions.mediation, scopeAndCrossOriginParent, WTFMove(callback));
```

Comment:
similar to above, let's capture the securityOrigin directly instead of capturing document and getting it in the lambda.